src/systems/systems/src/hydraulic/linear_actuator.rs

```diff
@@ -0,0 +1,707 @@
+use uom::si::{
+    area::square_meter,
+    f64::*,
+    force::newton,
+    length::meter,
+    ratio::ratio,
+    velocity::meter_per_second,
+    volume::{cubic_meter, gallon},
+    volume_rate::gallon_per_second,
+};
+
+use super::rigid_body::RigidBodyOnHingeAxis;
+
+use crate::simulation::UpdateContext;
+
+pub trait Actuator {
+    fn used_volume(&self) -> Volume;
+    fn reservoir_return(&self) -> Volume;
+    fn reset_accumulators(&mut self);
+}
+
+#[derive(PartialEq, Clone, Copy)]
+pub enum LinearActuatorMode {
+    ClosedValves = 0,
+    PositionControl = 1,
+    Damping = 2,
+}
+
+// LinearActuator represents a classical linear actuator with a rod side area and a bore side area
+// It is connected between an anchor point on the plane and a control arm of a rigid body
+// When the actuator moves, it takes fluid on one side and gives back to reservoir the fluid on other side
+// Difference of volume between both side will cause variation of loop reservoir level.
+// It moves between a max absolute and minimum absolute position. The position is finally normalized from 0 to 1 (compressed to extended)
+//
+// It can behave it two main ways: its control valves are either closed, and it can't move, or valves are opened and
+// hydraulic power can move it with enough pressure.
+//
+// How those two modes are done: when valves are closed, actuator is constrained by a spring damper system,
+// which simulate the fluid compressibility resisting to rigid body movement.
+// When valves are opened, a controller will try to reach the specified target position. Controller works in flow
+// control so only the correct amount of flow is sent to the actuator.
+#[derive(PartialEq, Clone, Copy)]
+pub struct LinearActuator {
+    number_of_actuators: u8,
+
+    position_normalized: Ratio,
+    position: Length,
+    last_position: Length,
+
+    speed: Velocity,
+    force: Force,
+
+    max_position: Length,
+    min_position: Length,
+
+    total_travel: Length,
+
+    bore_side_area: Area,
+    bore_side_volume: Volume,
+
+    rod_side_area: Area,
+    rod_side_volume: Volume,
+
+    volume_extension_ratio: Ratio,
+    signed_flow: VolumeRate,
+    flow_error_prev: VolumeRate,
+
+    max_flow: VolumeRate,
+    min_flow: VolumeRate,
+
+    delta_displacement: Length,
+
+    volume_to_actuator_accumulator: Volume,
+    volume_to_res_accumulator: Volume,
+
+    mode: LinearActuatorMode,
+    is_control_valves_closed: bool,
+    closed_valves_reference_position: Ratio,
+    opened_valves_target_position: Ratio,
+
+    fluid_compression_spring_constant: f64,
+    fluid_compression_damping_constant: f64,
+    active_damping_constant: f64,
+}
+impl LinearActuator {
+    const DEFAULT_I_GAIN: f64 = 0.2;
+    const DEFAULT_P_GAIN: f64 = 0.05;
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        connected_body: &RigidBodyOnHingeAxis,
```

Comment:
I had the same concern indeed, and it's difficult for me to truly separate those two. In fact if you need a min and max linear distance... that would only be for a linear actuator. Other way for an actuator to be connected would be maybe by a direct torque application, where in that case you wouldn't probably need that min/max distance.  Don't know what to do :D :D

The fact the actuator "asks" for min and max is that this way, it will always have the correct dimentions depending on the actuated body dimensions.

Reply:
> In fact if you need a min and max linear distance... that would only be for a linear actuator.

Ok, I see that. But if I understand it correctly, we would only be modifying the `LinearActuator` to accept a more generic rigid body, so it's fine if this concept of max and min only applies to it, no?

In this case, my second solution of accepting `max_pos` and `min_pos` as parameters instead of `RigidBodyOnHingeAxis` would still make it a bit easier to test this code as well, since we don't have to instantiate a rigid body every time we create a `LinearActuator`.

Anyways, I don't think this too important, since you do elaborate on the reason for the tight coupling between the actuator and the body later on, so separating them more here wouldn't really matter too much. 😄